cmd/kube-controller-manager/app/core.go

```diff
@@ -747,6 +748,31 @@ func startPersistentVolumeProtectionController(ctx context.Context, controllerCo
 	return nil, true, nil
 }
 
+func newVolumeAttributesClassProtectionControllerDescriptor() *ControllerDescriptor {
+	return &ControllerDescriptor{
+		name:     names.VolumeAttributesClassProtectionController,
+		initFunc: startVolumeAttributesClassProtectionController,
+		requiredFeatureGates: []featuregate.Feature{
+			features.VolumeAttributesClass,
+		},
+	}
+}
+
+func startVolumeAttributesClassProtectionController(ctx context.Context, controllerContext ControllerContext, controllerName string) (controller.Interface, bool, error) {
+	vacProtectionController, err := vacprotection.NewVACProtectionController(
+		klog.FromContext(ctx),
+		controllerContext.ClientBuilder.ClientOrDie("vac-protection-controller"),
+		controllerContext.InformerFactory.Core().V1().PersistentVolumeClaims(),
+		controllerContext.InformerFactory.Core().V1().PersistentVolumes(),
+		controllerContext.InformerFactory.Storage().V1alpha1().VolumeAttributesClasses(),
```

Comment:
other controllers started successfully but this controller failed to start due to missing resources.

```
sudo FEATURE_GATES=VolumeAttributesClass=true ./hack/local-up-cluster.sh
```
output: [kube-controller-manager.log](https://github.com/user-attachments/files/16189929/kube-controller-manager.log) It's very noisy for users.

Reply:
I think we will want to detect if the API is enabled, and skip the controller initialization if so. I'm asking around if there's an API to do that.

Reply:
>  detect if the API is enabled, and skip the controller initialization

This behavior has been removed, so it may not be recommended. https://github.com/kubernetes/kubernetes/pull/117977 @msau42

Reply:
some discussions: https://kubernetes.slack.com/archives/C0EG7JC6T/p1718249135862569

Reply:
Thanks, it looks like the recommendation now is to disable the feature gate if the api is also disabled.